mws/sitesmanagement/views/domains.py

```diff
@@ -96,7 +97,7 @@ def add_domain(request, vhost_id, socket_error=None):
                                     vhost.main_domain.name == vhost.service.network_configuration.name:
                         vhost.main_domain = new_domain
                         vhost.save()
-                launch_ansible(vhost.service)  # to add the new domain name to the vhost apache configuration
```

Comment:
if ansible is not executed when a new domain is added in the .usertest.mws3.csx.cam.ac.uk and the user has multiple vhosts, it will go to the default vhost and not the one with the domain. I think launch_ansible needs to be executed always.

Reply:
A request for an ACME certificate will fail while the domain is not visible in DNS. Ansible will be run through the DomainName.accept_it method for domains that need authorisation and ones in .usertest.mws3.csx once the name is plausibly in DNS.

Reply:
The user can still execute ansible by other means, like updating the list of users or adding a php module. Thus, I think the task to request an ACME certificate should rely on an AnsibleConfiguration object that would contain if the domain has been set up or not yet.

Reply:
Fair point about the user being able to launch Ansible, but I disagree with adding AnsibleConfiguration objects as there's currently no way to test for those from Ansible. Couldn't we just set the service to 'ansible_queued' and block further Ansible runs until certificate configuration is complete?